### PR TITLE
Feature: Loop over known AWS credentials for S3

### PR DESCRIPTION
**Pull request by josephsnyder**
When connecting to an S3 mirror, loop over the connection profiles
found on the system to determine which can be used to connect.

### CODE REVIEW

**Review by zackgalbreath — DISMISSED** [review]

**Comment by bernhardkaindl** [comment]
@josephsnyder codecov/patch failed, we can't merge this with a normal maintainer-merge.

**Comment by alalazo** [comment]
Can we close this PR, or is it still relevant?

**Comment by haampie** [comment]
It's not

**Pull request closed without merging**
